pandas/core/frame.py

```diff
@@ -6244,12 +6244,14 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         if not self.columns.is_unique:
             raise ValueError("columns must be unique")
 
-        result = self[column].explode()
-        return (
-            self.drop([column], axis=1)
-            .join(result)
-            .reindex(columns=self.columns, copy=False)
-        )
+        result = self.copy()
+        exploded_col = result.pop(column).reset_index(drop=True).explode()
+        result = result.reset_index().join(exploded_col)
+        result.set_index(result.columns[: self.index.nlevels].tolist(), inplace=True)
```

Comment:
you don't need to do all of this, try
```result.index = self.index.take(result.index)```

Reply:
Nice, I didn't know about `take`, thanks Jeff!